features/support/parameter_types.rb

```diff
@@ -184,7 +184,14 @@ def provider_by_name(name)
 ParameterType(
   name: 'metric',
   regexp: /metric "([^"]*)"/,
-  transformer: ->(name) { Metric.find_by!(system_name: name) }
+  transformer: ->(name) {
+    metrics = Metric.where(parent_id: nil)
+    metric = metrics.find_by(friendly_name: name) || metrics.find_by(system_name: name)
```

Comment:
Searching across all metrics in the table by system_name is less prone to finding the wrong one, than by friendly_name, yet even so not risk free. This seems to increase the risk of fetching the wrong metric when there are other tests running in parallel on the same db. Do we really want it to be unscoped?

Reply:
I just think it's far more readable to use the friendly name in the scenarios. I wouldn't call it risky but... 🤷🏻‍♂️

Reply:
@akostadinov  @hallelujah WDYT?

Reply:
I think this one is fine. There should not be other tests running in parallel on the same table, there were parallel tests before and they are scoped on DB name level.

On new Rails version it is also the case https://guides.rubyonrails.org/testing.html#parallel-testing-with-processes

> When parallelizing tests, Active Record automatically handles creating a database and loading the schema into the database for each process. The databases will be **suffixed with the number corresponding to the worker.** For example, if you have 2 workers the tests will create test-database-0 and test-database-1 respectively.